packages/vue-apollo-composable/src/useMutation.ts

```diff
@@ -118,7 +118,7 @@ export function useMutation<
     return null
   }
 
-  vm && onBeforeUnmount(() => {
+  vm && onScopeDispose(() => {
```

Comment:
Maybe we don't need the `vm` check anymore?

Reply:
I think this will cause `[Vue warn] onScopeDispose() is called when there is no active effect scope to be associated with.` if called outside of effect scope.

Would be probably nice to add a warning for missing `vm`, recommending a manual disposal.